src/lib/y2packager/clients/inst_productsources.rb

```diff
@@ -377,23 +372,17 @@ def NormalizeURL(url_string)
     def IsAddOnAlreadySelected(s_url, s_path)
       ret = -1
 
-      s_url = NormalizeURL(s_url)
+      search_url = NormalizeURL((s_url || "") + (s_path || ""))
+      # replace $releasever by the real version
+      search_url.gsub!("$releasever", Product.version)
```

Comment:
Use [Pkg.ExpandedUrl](
https://github.com/yast/yast-pkg-bindings/blob/78f5ce710aa8973f50227150ed570467d0d71885/src/PkgFunctions.cc#L514) to get the full URL. It uses the libzypp functionality.

It's not that trivial as it looks at the first sight. There are also other variables like `$arch`, `$releasever_major`, `$releasever_minor` and others. There might be even custom defined values. See the `Variable substitution` section in `man zypper`.